File: LC/3507_Minimum_Pair_Removal_to_Sort_Array_I/solution.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        total_operation = 0
        while True:
            if sorted(nums) == nums:
                break
            
            min_sum = inf
            min_index = 0
            for i in range(len(nums)-1):
                tmp = nums[i] + nums[i+1]
                if tmp < min_sum:
                    min_sum = tmp
                    min_index = i

            nums[min_index] = min_sum
            del nums[min_index+1]
            total_operation += 1

        return total_operation

File: LC/3507_Minimum_Pair_Removal_to_Sort_Array_I/solution.cpp
class Solution {
public:
    int minimumPairRemoval(vector<int>& nums) {
        int operations = 0;
        while (true) {
            bool is_sorted = true;
            for (int i = 0; i < nums.size()-1; i++) {
                if (nums[i] > nums[i+1]) {
                    is_sorted = false;
                    break;
                }
            }

            if (is_sorted) {
                return operations;
            }

            int min_pair_sum = INT_MAX;
            int min_pair_idx = 0;
            int tmp = 0;
            for (int i = 0; i < nums.size()-1; i++) {
                tmp = nums[i] + nums[i+1];
                if (tmp < min_pair_sum) {
                    min_pair_sum = tmp;
                    min_pair_idx = i;
                }
            }

            nums[min_pair_idx] = min_pair_sum;
            nums.erase(nums.begin() + min_pair_idx + 1);
            operations++;
        }
    }
}; 